edb/schema/constraints.py

```diff
@@ -679,15 +859,28 @@ class AlterConstraint(ConstraintCommand,
     referenced_astnode = qlast.AlterConcreteConstraint
 
     @classmethod
-    def _cmd_tree_from_ast(cls, schema, astnode, context):
+    def _cmd_tree_from_ast(
+        cls,
+        schema: s_schema.Schema,
+        astnode: qlast.DDLOperation,
+        context: sd.CommandContext,
+    ) -> AlterConstraint:
         cmd = super()._cmd_tree_from_ast(schema, astnode, context)
+        assert isinstance(cmd, AlterConstraint)
 
         if isinstance(astnode, (qlast.CreateConcreteConstraint,
                                 qlast.AlterConcreteConstraint)):
-            subject_ctx = context.get(ConsistencySubjectCommandContext)
+            # TODO: how to make the following in a more "type-compliant" way?
+            # ConsistencySubjectCommandContext is an empty mixin used to
+            # group some classes of ObjectCommand
+            subject_ctx = context \
+                .get(ConsistencySubjectCommandContext)  # type: ignore
```

Comment:
As in the "TODO" I wrote above. The empty mixin is used to group some types of ObjectCommand, I would like your guidance on how to refactor the code to make mypy happy and respect the original purpose of the code. 

As a side note, I understand well the "error" reported by mypy and its ground. At the same time, I think this use of empty mixin is even elegant and part of the beauty of Python (it looks a pity to fix!).

Reply:
There's no reason to keep `ConsistencySubjectCommandContext` a pure mixin, make it inherit from `ObjectCommandContext`.

Reply:
@elprans, when I modify the `ConsistencySubjectCommandContext` this way:

```py
class ConsistencySubjectCommandContext(
    sd.ObjectCommandContext[ConsistencySubject]
):
    # context mixin
    pass
```

mypy complains: "Value of type variable "CommandContextToken_T" of "get" of "CommandContext" cannot be "ConsistencySubjectCommandContext"mypy(error)"

I read the code several times, and I don't see why mypy comes to that conclusion.

1. the ConsistencySubject is a so.Object
2. ObjectCommandContext is CommandContextToken[ObjectCommand[so.Object_T]]
3. CommandContext.get expects a type of CommandContextToken_T

Can you please give me an hint on this?

Reply:
Smells like a mypy bug. Let's keep things as is.